### PR TITLE
Fixes JanusGraph#1281 that was causing removeLogProcessor (shutdown on the log processor) to not work as expected

### LINKED ISSUE #1281: LogProcessorFramework `removeLogProcessor`/`shutdown` not working
## Description
LogProcessorFramework has method **removeLogProcessor**, which takes the log identifier as the parameter. This however is neither stopping nor removing the log processor. 

Possible reason could be that the `Map processorLogs` in StandardLogProcessorFramework.java, which is being used to remove the logprocessor, was never actually populated while building the log processor. 

## How to reproduce?
* Add a log processor with some log processor identifier
* Do some operation in janusgraph to see that the log processor is functional
* remove the logprocessor using removeLogProcessor method
* Do a operation again to find out that the log processor was not removed

### PR DESCRIPTION
**Pull request by codemonk5**
Fixes #1281

Signed-off-by: Prakhar Agrawal <Prakhar.x.Agrawal@gs.com>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you written and/or updated unit tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [x] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?
- [x] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: janusgraph-test/src/test/java/org/janusgraph/graphdb/log/StandardLogProcessorFrameworkTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
+package org.janusgraph.graphdb.log;
```
Comment (farodin91) [inline janusgraph-test/src/test/java/org/janusgraph/graphdb/log/StandardLogProcessorFrameworkTest.java]: You forgot to add the copyright.

**Comment by porunov** [comment]
@codemonk5 We have switched to automated CLA system. You should sign the CLA again by using the link https://api.lfcla.com/v2/repository-provider/github/sign/1543822/77385607/1742
It takes very short time to sign it online

File: janusgraph-test/src/test/java/org/janusgraph/graphdb/log/StandardLogProcessorFrameworkTest.java (hunk 1 above)
Comment (codemonk5) [reply janusgraph-test/src/test/java/org/janusgraph/graphdb/log/StandardLogProcessorFrameworkTest.java]: Added it in the new commit.

**Comment by porunov** [comment]
@codemonk5 For CCLA you can follow the next instructions: https://github.com/swinslow/cla-tool-docs/blob/master/walkthroughs/3-Corporate-Contributor-first-for-company.md

File: janusgraph-test/src/test/java/org/janusgraph/graphdb/log/StandardLogProcessorFrameworkTest.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        logProcessorFramework.addLogProcessor("foo-identifier")
+            .addProcessor((tx, txId, changeState) -> {
+                // no-op processor
+            })
+            .setStartTimeNow()
+            .setProcessorIdentifier("bar-processor")
+            .build();
+
+        Map<String, Log> processorLogs = null;
+        try {
```
Comment (FlorianHockmann) [inline janusgraph-test/src/test/java/org/janusgraph/graphdb/log/StandardLogProcessorFrameworkTest.java]: Can't we omit this try/catch block? I think tests should be as simple as possible and the tests also worked when I removed this and instead added the exceptions to the method signature.

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/log/StandardLogProcessorFramework.java [hunk 3]
```diff
@@ -186,6 +186,7 @@ public MessageReader apply(@Nullable ChangeProcessor changeProcessor) {
                             return new MsgReaderConverter(userLogName, changeProcessor, retryAttempts);
                         }
                     }));
+                    processorLogs.put(userLogName,log);
```
Comment (FlorianHockmann) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/log/StandardLogProcessorFramework.java]: Really a tiny nit, but could you please add a space after the comma?

**Comment by porunov** [comment]
@codemonk5 also, could you please retarget this PR to `v0.3` branch please?

**Comment by mbrukman** [comment]
@codemonk5 --- were you going to replace this PR as suggested by @porunov?

FWIW, I fixed (another) bug in the @janusgraph-bot code so this PR is now correctly marked as `[cla: yes]`, based on the original paper-based CLA (sorry about that!). Please also add yourself to your employer's CCLA via the EasyCLA link above, so that this is handled correctly going forward and in the future when we decommission @janusgraph-bot.

**Comment by porunov** [comment]
@codemonk5 the support for `0.3` and `0.4` branches are now dropped. We are targeting `master` branch (`0.5.0` release). So, you are not required to retarget your PR to other branches. Could you just resolve comments from @FlorianHockmann ?

**Comment by porunov** [comment]
Fixed in #2014

**Pull request closed without merging**